include/eve/module/real/core/function/regular/generic/if_else.hpp

```diff
@@ -90,9 +104,7 @@ namespace eve::detail
   //------------------------------------------------------------------------------------------------
   // Optimizes if_else(c,t,constant)
   template<value T, value U, typename Constant>
-  requires( ( kumi::product_type<U> && std::is_invocable_v<Constant, as<kumi::element_t<0,U>>> )
-          || std::is_invocable_v<Constant, as<U>>
-          )
+  requires( !kumi::product_type<Constant> && std::invocable<Constant, as<first_of_t<U>>> )
```

Comment:
Why only first_of?

Reply:
I have no easy way here to iterate over all the member at this place. I can add a traits doing this if needed.

Reply:
Ah. Well whatever.

`kumi::element_t<0, T>`?